src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1926,4 +1942,27 @@ public static String getVersionName() {
         // value
         return "1.1.12";
     }
+
+    /**
+     * A {@link Future}  whose result can be set by a {@link #set(Object)} or {@link #setException(Throwable)}
+     */
+    final static class SettableFuture<V> extends FutureTask<V> {
```

Comment:
If that works it sounds like a great idea.  It would be easy to read and only affects the issue at hand.